apps/backend/src/modules/activity-logs/activity-log.repository.ts

```diff
@@ -42,23 +42,26 @@ class ActivityLogRepository implements Repository {
 
 	public async findAll({
 		endDate,
-		projectId,
+		hasRootPermission,
+		projectIds,
 		startDate,
-	}: ActivityLogQueryParameters): Promise<{ items: ActivityLogEntity[] }> {
+	}: {
+		hasRootPermission: boolean;
+		projectIds: number[];
+	} & ActivityLogQueryParameters): Promise<{ items: ActivityLogEntity[] }> {
 		const query = this.activityLogModel
 			.query()
 			.withGraphFetched("[gitEmail.contributor, project, createdByUser]")
 			.modifyGraph("gitEmail.contributor", (builder) => {
 				builder.select("id", "name");
 			})
-			.whereBetween("activity_logs.date", [startDate, endDate])
-			.orderBy("date");
+			.whereBetween("activity_logs.date", [startDate, endDate]);
 
-		if (projectId) {
-			query.where("project_id", projectId);
+		if (!hasRootPermission) {
+			query.whereIn("activity_logs.projectId", projectIds);
 		}
```

Comment:
I'm not sure but seems like this logic should be in service. Like one method in repository to retrieve for user with root permission and another to retrieve with project id's.